security/webauthn/src/test/java/AbstractWebAuthnTest.java

```diff
@@ -0,0 +1,215 @@
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.restassured.RestAssured;
+import io.restassured.filter.Filter;
+import io.restassured.http.ContentType;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+import io.vertx.core.json.JsonObject;
+
+public abstract class AbstractWebAuthnTest {
+
+    protected abstract RestService getApp();
+
+    private static final String REGISTER_URL = "/q/webauthn/register";
+    private static final String REGISTER_CALLBACK_URL = "/q/webauthn/callback";
+    private static final String LOGIN_URL = "/q/webauthn/login";
+    private static final String LOGOUT_URL = "/q/webauthn/logout";
+
+    private static final String PUBLIC_API_URL = "/api/public";
+    private static final String PUBLIC_ME_API_URL = "/api/public/me";
+    private static final String USER_API_URL = "/api/users/me";
+    private static final String ADMIN_API_URL = "/api/admin";
+
+    private static final String USERNAME = "Roosvelt";
+
+    private static Filter cookieFilter;
+
+    enum User {
+        USER,
+        ADMIN;
+    }
+
+    @BeforeAll
+    public static void setup() {
+        cookieFilter = new CookieFilter();
```

Comment:
Through this `cookieFilter` unit tests share state, but test class methods are not ordered. Please enforce method order so that we have explicitly set expected order of execution.

Reply:
My point is to guarantee that added tests are idempotent and expected state at each method execution is clear.

Reply:
I've included @TestMethodOrder(MethodOrderer.OrderAnnotation.class) at level class on AbstractWebAuthnTest.
And also @Order() test methods.